code/src/main/java/habit_mode/view_model/HabitViewModel.java

```diff
@@ -197,6 +197,28 @@ public void removeHabit(Habit habitToRemove) {
 
     }
 
+    /**
+     * Updates the currently selected habit
+     * 
+     * @precondition index >= 0 & index < this.habitListProperty().getValue().size();
+     * @postcondition this.selectedHabitProperty().getText() == this.removeHabitNameProperty()
+     *                this.selectedHabitProperty().getFrequency() == the selected frequency property;
+     * 
+     * @param index the index of the habit to update.
+     */
+    public void updateHabit(int index) {
+        if (index < 0) {
+            throw new IllegalArgumentException("the index cannot be less than 0");
+        }
+        if (index > this.habitListProperty.getValue().size()) {
+            throw new IllegalArgumentException("the index cannot be greater than the size of the habit list");
+        }
+        Habit updatedHabit = new Habit(this.removeHabitNameProperty().getValue(), this.determineRemoveFrequency());
+        this.removeHabit(this.habitListProperty.getValue().get(index));
+        this.habitListProperty.getValue().add(index, updatedHabit);
+        this.getServerCommunicator().addHabit(updatedHabit);
```

Comment:
We have an option for updating habits on the server.
However, since it hasn't been implemented with the ServerServerCommunicator, we are unable to send the request. I will raise this as an issue to be fixed in the future.